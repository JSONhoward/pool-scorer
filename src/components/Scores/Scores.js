import React from 'react'
import styled from 'styled-components'

const ScoresBox = styled.div`
display: flex;
width: fit-content;
justify-content: space-between;
margin-bottom: 1rem;
`

const CurrentScore = styled.p`
display: grid;
place-items: center;
min-width: 5rem;
font-size: 5rem;
opacity: ${props => !props.scoreOpacity ? '1' : '.5'};
border: 2px solid white;
margin: 0 5px;
border-radius: 10px;
`

const Scores = ({ players, score1, score2, player1 }) => {
    return (
        <ScoresBox>
            {
                players === 1 ? (<CurrentScore>{score1}</CurrentScore>) :
                    (
                        <>
                            <CurrentScore scoreOpacity={!player1}>{score1}</CurrentScore>
                            <CurrentScore scoreOpacity={player1}>{score2}</CurrentScore>
                        </>
                    )
            }
        </ScoresBox>
    )
}

export default Scores
