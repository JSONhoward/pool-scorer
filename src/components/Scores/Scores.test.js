import React from 'react'
import {create} from 'react-test-renderer'
import Scores from './Scores'

describe('Scores Tests', () => {
    test('Snapshot Test', () => {
        const tree = create(<Scores />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Receives Props', () => {
        const props = {
            players: 2,
            score1: 5,
            score2: 7,
            player1: true,
            match: false,
            raceTo: 11,
            gameOver: false
        }

        const component = create(<Scores {...props} />).root

        expect(component.props).toEqual(props)
    })
})