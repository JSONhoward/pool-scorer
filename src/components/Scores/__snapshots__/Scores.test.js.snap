// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scores Tests Snapshot Test 1`] = `
<div
  className="sc-AxjAm iwecYv"
>
  <p
    className="sc-AxirZ cBszJp"
  />
  <p
    className="sc-AxirZ foamJh"
  />
</div>
`;
